### PR TITLE
[MAINT] add lint recipe to make file

### LINKED ISSUE #3756: add make recipe to run all linting tool at once
add a recipe in the make file to run all linting tools with their proper config

this would provide an 'alternative' for people who have access to make and do not use pre-commit

### PR DESCRIPTION
**Pull request by Remi-Gau**
Closes #3756

Changes proposed in this pull request:

- add lint recipe to make file
- move linting dependencies to the [dev] install group

### CODE REVIEW

**Comment by Remi-Gau** [comment]
complete after #3759 is merged

File: Makefile [hunk 1]
```diff
@@ -64,3 +64,12 @@ doc:
 .PHONY : pdf
 pdf:
 	make -C doc pdf
+
+.PHONY : lint
+# TODO: add flynt when f strings are implemented in the whole codebase
```
Comment (ymzayek) [inline Makefile:69]: This note can be removed now?

Comment (Remi-Gau) [reply Makefile]: yup. thanks for catching it.

**Review by ymzayek — APPROVED** [review]
Great thx!

**Pull request merged**